automation_tools/__init__.py

```diff
@@ -1922,20 +1922,39 @@ def host_pings(host, attempts=200):
         output = command.communicate()[0]
         # Checking the return code of ping is 0
         if int(output.split()[-1]) == 0:
+            print(
+                'SUCCESS !! The given host {0} has been pinged!!'.format(host))
             break
+        elif time.time() > timeup:
+            print(
+                'The timout for pinging the host {0} has reached!'.format(host)
+            )
+            sys.exit(1)
         else:
             time.sleep(5)
 
 
-def get_hostname_from_ip(ip):
+def get_hostname_from_ip(ip, timeout=3):
     """Retrives the hostname by logging into remote machine by IP.
     Specially for the systems who doesnt support reverse DNS.
     e.g usersys machines.
 
     :param ip: A string. The IP address of the remote host.
+    :param timeout: An integer. The polling timeout in minutes.
```

Comment:
All over the upgrade automation, I have followed current style and need to change it all over.
So I will be raising new PR for that. As of now you may merge this PR.